### PR TITLE
Clean up `synapse.rest.admin`

### PR DESCRIPTION
**Pull request by dklimpel**
No functional changes
Only code cleanup

- harmonize `PATTERNS`
  - there are `PATTERNS` with and without trailing `$`
  - there are `PATTERNS` with `[^/]+` and `[^/]*`
- replace
  ```
  requester = await self._auth.get_user_by_req(request)
  await assert_user_is_admin(self._auth, requester.user)
  ```
  with `await assert_requester_is_admin(self._auth, request)` if `requester` is not needed
- remove not needed `self.hs = hs`
- use `allowed_values` of `parse_string`

### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#changelog). The entry should:
  - Be a short description of your change which makes sense to users. "Fixed a bug that prevented receiving messages from other servers." instead of "Moved X method from `EventStore` to `EventWorkerStore`.".
  - Use markdown where necessary, mostly for `code blocks`.
  - End with either a period (.) or an exclamation mark (!).
  - Start with a capital letter.
* [x] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
* [x] [Code style](https://matrix-org.github.io/synapse/latest/code_style.html) is correct
  (run the [linters](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#run-the-linters))

Signed-off-by: Dirk Klimpel dirk@klimpel.org

### CODE REVIEW

File: tests/rest/admin/test_statistics.py [hunk 1]
```diff
@@ -92,7 +92,7 @@ def test_invalid_parameter(self) -> None:
             channel.code,
             msg=channel.json_body,
         )
-        self.assertEqual(Codes.INVALID_PARAM, channel.json_body["errcode"])
+        self.assertEqual(Codes.UNKNOWN, channel.json_body["errcode"])
```
Comment (squahtx) [inline tests/rest/admin/test_statistics.py:95]: It's unfortunate that the errcode is now `UNKNOWN`. Is it something that can be fixed easily?

**Review by squahtx — APPROVED** [review]
lgtm, one question

File: tests/rest/admin/test_statistics.py (hunk 1 above)
Comment (dklimpel) [reply tests/rest/admin/test_statistics.py:95]: Before, the error itself was generated in `statistics.py` and now the already existing function  `parse_string` is used. There it is no errcode implemented.

https://github.com/matrix-org/synapse/blob/e8ae94a22367a81049582dfdb16c743a45ca4e8b/synapse/http/servlet.py#L383-L388

Comment (squahtx) [reply tests/rest/admin/test_statistics.py:95]: I see!

Would it make sense to set the errcode for the SynapseError?
```py
raise SynapseError(400, message, Codes.INVALID_PARAM)
```

Comment (dklimpel) [reply tests/rest/admin/test_statistics.py:95]: IMO, yes. But this should be a new issue / PR. Probably there are a lot of changes to the unit tests.

**Pull request merged**